### PR TITLE
reduce positioning operation to single operation

### PR DESCRIPTION
**Pull request by walterddr**
instead of positioning and seek to read value. This PR reduce the operation into direct access.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java [hunk 1]
```diff
@@ -211,7 +211,7 @@ public BaseDataBlock(ByteBuffer byteBuffer)
    * @param rowId row ID
    * @param colId column ID
    */
-  protected abstract void positionCursorInFixSizedBuffer(int rowId, int colId);
+  protected abstract int computePositionCursorInFixSizedBuffer(int rowId, int colId);
```
Comment (siddharthteotia) [inline pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java:214]: (nit) update the jaavadoc to include the return value please ?

**Review by siddharthteotia — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java [hunk 2]
```diff
@@ -158,26 +158,22 @@ public int getNumberOfRows() {
...
   }
 
   @Override
   public double getDouble(int rowId, int colId) {
-    _fixedSizeData.position(rowId * _rowSizeInBytes + _columnOffsets[colId]);
-    return _fixedSizeData.getDouble();
+    return _fixedSizeData.getDouble(rowId * _rowSizeInBytes + _columnOffsets[colId]);
   }
 
   @Override
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java:179]: Let's also fix the `getString()`

File: pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java:214]: Suggest renaming it to `getOffsetInFixedBuffer`

File: pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java [hunk 3]
```diff
@@ -221,7 +221,7 @@ public BaseDataBlock(ByteBuffer byteBuffer)
    * @param colId column ID
    * @return the length to extract from variable size buffer.
    */
-  protected abstract int positionCursorInVariableBuffer(int rowId, int colId);
+  protected abstract int computePositionCursorInVariableBuffer(int rowId, int colId);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java:224]: Don't change this method name. It does not return the offset, but actually position the cursor

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java (hunk 2 above)
Comment (walterddr) [reply pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java:179]: yeah dont know why I missed that.

File: pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java (hunk 3 above)
Comment (walterddr) [reply pinot-core/src/main/java/org/apache/pinot/core/common/datablock/BaseDataBlock.java:224]: yeah and return the length to read out of the var-byte buffer. let me rethink the naming

**Pull request merged**